sr_grasp/src/sr_grasp/utils.py

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env python
+
+# ##########################################################################
+# Copyright (c) 2014 Shadow Robot Company Ltd.
+#
+# This program is free software: you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by the Free
+# Software Foundation, either version 2 of the License, or (at your option)
+# any later version.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+# ###########################################################################
+
+import rospy
+from moveit_msgs.msg import Grasp
+from trajectory_msgs.msg import JointTrajectoryPoint
+
+_sr_joint_names = [
+        'FFJ1', 'FFJ2', 'FFJ3', 'FFJ4',
+        'LFJ1', 'LFJ2', 'LFJ3', 'LFJ4', 'LFJ5',
+        'MFJ1', 'MFJ2', 'MFJ3', 'MFJ4',
+        'RFJ1', 'RFJ2', 'RFJ3', 'RFJ4',
+        'THJ1', 'THJ2', 'THJ3', 'THJ4', 'THJ5',
+        'WRJ1', 'WRJ2']
+
+_sr_joint_names_j0 = [
+        'LFJ0', 'LFJ3', 'LFJ4', 'LFJ5',
+        'RFJ0', 'RFJ3', 'RFJ4',
+        'MFJ0', 'MFJ3', 'MFJ4',
+        'FFJ0', 'FFJ3', 'FFJ4',
+        'THJ1', 'THJ2', 'THJ3', 'THJ4', 'THJ5',
+        'WRJ1', 'WRJ2']
+
+def _fix_j0(joints):
+    """
+    Convert joints targets using J1 and J2 to J0, which the controllers use.
+    Useful if using joint_states to get grasp joint positions.
+    """
+    for finger in ['FFJ', 'LFJ', 'MFJ', 'RFJ']:
+        if finger+'1' in joints and finger+'2' in joints:
+            joints[finger+'0'] = joints[finger+'1'] + joints[finger+'2']
```

Comment:
Not using the traj controller atm, so leaving in for now. It wont be later.
Going to do traj control as a seperate pr if time.